src/e2e/app.ts

```diff
@@ -392,6 +401,90 @@ export class App {
     return downloadButton.click();
   }
 
+  async connect(): Promise<void> {
+    const document = await this.document();
+    const connectButton = await document.findByRole("button", {
+      name: "Connect",
+    });
+    await connectButton.click();
+    await document.findByRole("button", {
+      name: "Serial menu",
+    });
+  }
+
+  async disconnect(): Promise<void> {
+    const document = await this.document();
+    const disconnectButton = await document.findByRole("button", {
+      name: "Disconnect",
+    });
+    await disconnectButton.click();
+    return waitFor(
+      async () => {
+        expect(
+          await document.queryByRole("button", {
+            name: "Serial menu",
+          })
+        ).toBeNull();
+      },
+      {
+        onTimeout: () => new Error("Serial still present after disconnect"),
+      }
+    );
+  }
+
+  async serialShow(): Promise<void> {
+    const document = await this.document();
+    const showSerialButton = await document.findByRole("button", {
+      name: "Show serial",
+    });
+    await showSerialButton.click();
+    // Make sure the button has flipped.
+    await document.findByRole("button", {
+      name: "Hide serial",
+    });
+  }
+
+  async serialHide(): Promise<void> {
+    const document = await this.document();
+    const hideSerialButton = await document.findByRole("button", {
+      name: "Hide serial",
+    });
+    await hideSerialButton.click();
+    // Make sure the button has flipped.
+    await document.findByRole("button", {
+      name: "Show serial",
+    });
+  }
+
+  async findSerialCompactTraceback(text: string): Promise<void> {
+    const document = await this.document();
+    await document.findByText(/SyntaxError/);
+    //await document.findByRole("link", {
+    //  name: text,
+    //});
```

Comment:
This can probably use the commented out version now. Either that or we change the signature.